corehq/apps/app_manager/views/releases.py

```diff
@@ -650,3 +665,39 @@ def toggle_build_profile(request, domain, build_id, build_profile_id):
                 build.build_profiles[build_profile_id].name
             ))
     return HttpResponseRedirect(reverse('download_index', args=[domain, build_id]))
+
+
+@require_deploy_apps
+def paginate_release_logs(request, domain, app_id):
+    limit = request.GET.get('limit')
+    page = int(request.GET.get('page', 1))
+    page = max(page, 1)
+    try:
+        limit = int(limit)
+    except (TypeError, ValueError):
+        limit = DEFAULT_PAGE_LIMIT
+
+    app_release_logs = ApplicationReleaseLog.objects.filter(app_id=app_id).order_by('-created_at')
+    paginator = Paginator(object_list=app_release_logs, per_page=limit)
+    current_page = paginator.get_page(page)
+
+    timezone = get_timezone_for_user(request.couch_user, domain)
+    transformed_logs = list(populate_data_app_release_logs(log, timezone) for log in current_page)
+
+    return JsonResponse({
+        'app_release_logs': transformed_logs,
+        'pagination': {
+            'total': paginator.count,
+            'num_pages': paginator.num_pages,
+            'current_page': page,
+            'more': page * limit < 10,
+        }
+    })
+
+
+def populate_data_app_release_logs(log, timezone):
+    timestamp = ServerTime(log.created_at).user_time(timezone)
```

Comment:
Can we pass this instead to this function? It seems like it would create this each time for a log.

Reply:
This calculates the time for when each log was created by accessing the `log.created_at` and then adjusting it according to the timezone of user.

https://github.com/dimagi/commcare-hq/blob/7e95cc4e99f67d7f4ea4f39588a0bd251bd608a1/corehq/apps/app_manager/views/releases.py#L685

This function is used here and adds additional data to response.